chainer/function_set.py

```diff
@@ -28,8 +31,20 @@ def __init__(self, **functions):
                 object as attributes.
 
         """
-        for name, func in six.iteritems(functions):
-            setattr(self, name, func)
+
+        for key, func in six.iteritems(functions):
+            if key == 'name':
```

Comment:
Excuse me, why do you need to fix `FunctionSet`?
We are now discussing about interface of `FunctionSet`. It may conflicts this change.